### ISSUE TITLE
minor implementation clarity improvement for readBody warning

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7802]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#7802|
|Type|enhancement|
|Created|2015-03-06 19:46:42Z|
|Branch|https://github.com/twisted/twisted/tree/readBody-None-transport-7802|

It's not quite enough to warrant a revert, but exarkun's comments on [#7762](https://github.com/twisted/twisted/issues/7762) should nevertheless be addressed.

<details><summary>Searchable metadata</summary>

```
trac-id__7802 7802
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_readBody_None_transport_7802 branches/readBody-None-transport-7802
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1425671202418762 1425671202418762
changetime__1425797513490527 1425797513490527
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [43974]) Branching to readBody-None-transport-7802.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @glyph|
|-|-|

Changes looks good for merge.

are there any guidelines for where to place inline function definitions, ex place them at the start of method parent method.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

Replying to [adiroiban](#note_3):
> Changes looks good for merge.

> are there any guidelines for where to place inline function definitions, ex place them at the start of method parent method.

Nope. This is a matter of taste for the author :-).

Generally, I like to define callbacks for Deferreds after the Deferred themselves, so that the code flow is in order of temporal execution.  In the case of cancellers, you have to declare them first, which in this case is unfortunate since the canceller references the protocol which has to be declared _after_ the Deferred is created, since there's a circular reference.  If I could have defined things in a different order here I probably would have.

> Thanks!

Thanks for the review!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set status to `closed`|
|-|-|

(In [43995]) Merge readBody-None-transport-7802: fix a brief issue.

Author: glyph

Reviewer: adiroiban

Fixes: [#7802](https://github.com/twisted/twisted/issues/7802)

Use 'is' to compare the transport, avoid doing the same "getattr" twice in
different ways.
